akka-actor/src/main/scala/akka/io/SelectionHandler.scala

```diff
@@ -303,9 +303,17 @@ private[io] class SelectionHandler(settings: SelectionHandlerSettings)
   private[this] var sequenceNumber = 0L // should be Long to prevent overflow
   private[this] var childCount = 0
   private[this] val registry = {
-    val dispatcher = context.system.dispatchers.lookup(SelectorDispatcher)
+    val dispatcher = context.system.dispatchers.lookup(SelectorDispatcher match {
+      case ""       => context.system.dispatchers.internalDispatcherId
+      case nonEmpty => nonEmpty
+    })
```

Comment:
can this be extracted to a new `lookup` method in Dispatchers?

Reply:
I don't know, I think it rather belongs in the settings logic, since it is about config semantics, but the `SelectionHandlerSettings` didn't have an `ActorSystem` in scope and is public so that's why I put it here to avoid bincomp issues, but maybe it is in effect internal?

Reply:
Would be nice to not copy these lines to all places. I was thinking that it belonged in Dispatchers  since the fallback if `""` is already there. I'm tempted to suggest that it should be a change in the existing `lookup`?

Reply:
As commented above, why can't we keep explicit values in those settings?